data_structures/static_rtree.hpp

```diff
@@ -642,11 +642,56 @@ class StaticRTree
         return result_coordinate.is_valid();
     }
 
+    /**
+     * Checks whether A is between B-range and B+range, all modulo 360
+     * e.g. A = 5, B = 5, range = 10 == true
+     *      A = -6, B = 5, range = 10 == false
+     *      A = -2, B = 355, range = 10 == true
+     *      A = 6, B = 355, range = 10 == false
+     *      A = 355, B = -2, range = 10 == true
+     *
+     * @param A the bearing to check, in degrees, 0-359, 0=north
+     * @param B the bearing to check against, in degrees, 0-359, 0=north
```

Comment:
The documentation states as precondition for `A` and `B` to be in the range 0-359 (both including I suppose?).
But then you do modulo calculation to bring `A` and `B` in this range in case it is outside of it.

Following that:

- either the precondition is wrong and I am allowed to pass in values outside of the stated range, or
- the precondition is right, then assert this instead of doing modulo calculation